pkg/fqdn/rulegen.go

```diff
@@ -145,6 +145,47 @@ perRule:
 	}
 }
 
+// AreFQDNRulesStillPresent is a function that validates that the rule UUID is
+// still present in the FQDN rules, it returns true if the rules do not have
+// FQDN rule or are still in the repo, returns false if any rule were deleted
+// from gen.AllRules.
+// The main reason is to avoid race conditions in the policyAdd, a new CNP
+// event can update the rule and delete the old ones, this prevents that an old
+// rule is deleted.
+// @TODO this need to be changed to use PolicyRepo when the operation can be
+// o(1) (using UUID) instead of O(n2)
+func (gen *RuleGen) AreFQDNRulesStillPresent(sourceRules []*api.Rule) bool {
+	gen.Lock()
+	defer gen.Unlock()
+
+	result := false
+	rulesHasFQDN := false
+	for _, sourceRule := range sourceRules {
+		if hasToFQDN(sourceRule) {
+			rulesHasFQDN = true
+			uuid := sourceRule.Labels.Get(uuidLabelSearchKey)
```

Comment:
I wanted to say that we can shortcut everything and just search for the UUID label only. We should leave a comment explaining that Labels.Get is expensive and that hasToFQDN is a quicker sanity check.

Reply:
I just wanted to make it sure that has a FQDN because maybe one day a UUID is added and all rules stop working ;-)